packages/core/src/__tests__/__e2e__/scrim/Scrim.cy.tsx

```diff
@@ -0,0 +1,278 @@
+        const parentRef = useRef<HTMLDivElement>(null);
+        const [open, setOpen] = useState(false);
+        return (
+          <div ref={parentRef}>
+            <Scrim open={open} containerRef={parentRef}>
+              <button onClick={() => setOpen((old) => !old)}>
+                CLOSE SCRIM
+              </button>
+            </Scrim>
+            <button
+              data-testid="openScrim"
+              onClick={() => setOpen((old) => !old)}
+            >
+              OPEN SCRIM
+            </button>
+            <input tabIndex={-1} type="text" />
+          </div>
+        );
+      }
+
+      cy.mount(<TestComponent />);
+
+      cy.findByRole("button", { name: "OPEN SCRIM" }).click();
+
+      cy.findByTestId("openScrim").should("have.attr", "tabindex", "-1");
+      cy.findByRole("textbox", { hidden: true }).should(
+        "have.attr",
+        "tabindex",
+        "-1"
+      );
+
+      cy.findByRole("button", { name: "CLOSE SCRIM" }).click();
+
+      cy.findByRole("button", { name: "OPEN SCRIM" }).should(
+        "not.have.attr",
+        "tabindex"
+      );
+      cy.findByRole("textbox").should("have.attr", "tabindex", "-1");
+    });
+  });
+
+  describe("WHEN disableFocusTrap is true", () => {
+    it("THEN should allow focus to leave the focus trap", () => {
+      function TestComponent() {
+        const [open, setOpen] = useState(false);
+        return (
+          <>
+            <Scrim open={open} disableFocusTrap>
+              <button onClick={() => setOpen((old) => !old)}>
+                CLOSE SCRIM
+              </button>
+            </Scrim>
+            <button onClick={() => setOpen((old) => !old)}>OPEN SCRIM</button>
+          </>
+        );
+      }
+      cy.mount(<TestComponent />);
+
+      cy.findByRole("button", { name: "OPEN SCRIM" }).click();
+      cy.findByRole("button", { name: "CLOSE SCRIM" }).should("have.focus");
+      cy.findByRole("button", { name: "CLOSE SCRIM" }).realPress("Tab");
+      // findByRole doesn't work here because FocusManager puts aria-hidden on all the elements outside its scope
+      cy.contains("button", "OPEN SCRIM").should("have.focus");
+    });
+  });
+
+  describe("WHEN disableReturnFocus is true", () => {
+    it("THEN should NOT return focus to the last focused element before", () => {
+      function TestComponent() {
+        const [open, setOpen] = useState(false);
+        return (
+          <>
+            <Scrim open={open} disableReturnFocus>
+              <button onClick={() => setOpen((old) => !old)}>
+                CLOSE SCRIM
+              </button>
+            </Scrim>
+            <button onClick={() => setOpen((old) => !old)}>OPEN SCRIM</button>
+          </>
+        );
+      }
+      cy.mount(<TestComponent />);
+
+      cy.findByRole("button", { name: "OPEN SCRIM" }).click();
+      cy.findByRole("button", { name: "CLOSE SCRIM" })
+        .should("have.focus")
+        .realClick();
+      cy.wait(50);
```

Comment:
Is the wait actually needed?

Reply:
Yeah, this one is a bit hard to test because if you removed the wait it would still pass iirc. But that would be a false positive because the return of the focus takes a bit of time so we don't want to check immediately.

Reply:
Not sure how it would pass as a false positive though?

Reply:
Because what we are trying to test here is the return of focus functionality, and that happens at a delay. If you test immediately, you are not really testing that functionality.

Reply:
But if it returns true the element has focus which is the correct behaviour? Cypress should manage the retry or the delay if needed?

Reply:
We are testing that the previous button does **not** have focus. If we assert that immediately it will return true, because it doesn't, and then cypress will move on but at the time of the assertion the return focus hasn't done its thing yet. There is probably a better way to test this that I didn't think of but just removing the `wait()` is not it. The way I made sure this test was making the right assertions was to toggle the return focus functionality on/off while commenting out the wait command.

Reply:
Tbh, I don't even remember the specifics of that scenario just that I did play around with this and confirmed that I need the wait. I don't like adding arbitrary waits in my tests but usually it is the best way to prove a negative in a test.

Reply:
Unless you have something else in the ui to wait for. Here I don't think I have. Maybe waiting for the Scrim to disappear but I'm not sure of the timings. 🤔